src/main/scala/scalafix/internal/sbt/ScalafixCoursier.scala

```diff
@@ -20,8 +20,8 @@ object ScalafixCoursier {
     BuildInfo.scalafixVersion
   )
 
-  def scalafixCliJars: List[Path] = {
-    CoursierSmall.fetch(fetchSettings.withDependencies(List(scalafixCli)))
+  def scalafixCliJars(repositories: Seq[Repository]): List[Path] = {
+    CoursierSmall.fetch(fetchSettings(repositories, List(scalafixCli)))
```

Comment:
Looking at this change again I'm not convinced this was necessary. The classpath fetched here should only contain scalafix-cli and not custom rules. It's only the "tool classpath" that should use custom resolvers. I had validated that custom resolvers worked as expected so I'm starting to believe something else is going on with your custom nexus repository, does it maybe require authentication?

Reply:
no, the Nexus repo in question doesn't require authentication

before this PR was merged, `scalafix` was downloading some (not all, but still some) artifacts from `repo1.maven.org`, even though `scalafixCustomResolvers` was configured to use only the internal Nexus mirror
see the screenshots in the linked issue as proof :wink: 

after this change, `scalafix` abides by the `scalafixCustomResolvers` (for all artifacts, not just some)

maybe my PR changes more things than necessary, I'm not sure, I'm not deeply familiar with the codebase. but some fix was necessary, because `scalafixCustomResolvers` wasn't working correctly before